mobile/lib/modules/asset_viewer/providers/image_viewer_page_state.provider.dart

```diff
@@ -42,9 +46,23 @@ class ImageViewerStateNotifier extends StateNotifier<ImageViewerPageState> {
 
     state = state.copyWith(downloadAssetStatus: DownloadAssetStatus.idle);
   }
+
+  void shareAsset(AssetResponseDto asset, BuildContext context) async {
+    showDialog(
+      context: context,
+      builder: (BuildContext buildContext) {
+        _shareService
+            .shareAsset(asset)
+            .then((_) => Navigator.pop(buildContext));

```

Comment:
Can we use `AutoRouter.of(context).pop()` to keep navigation within the `auto_router` package?

https://pub.dev/packages/auto_route

Reply:
This does not seem to work. DeleteDialog also uses `Navigator.of(buildContext).pop()`.

Reply:
Ah, ok then. We can leave it as it